doc/source/whatsnew/v1.1.0.rst

```diff
@@ -1068,6 +1068,7 @@ Sparse
 - Bug in :meth:`Series.sum` with ``SparseArray`` raises ``TypeError`` (:issue:`25777`)
 - Bug where :class:`DataFrame` containing :class:`SparseArray` filled with ``NaN`` when indexed by a list-like (:issue:`27781`, :issue:`29563`)
 - The repr of :class:`SparseDtype` now includes the repr of its ``fill_value`` attribute. Previously it used ``fill_value``'s  string representation (:issue:`34352`)
+- Fixed bug where :class:`DataFrame` or :class:`Series` could not be cast to :class:`SparseDtype` when empty (:issue:`33113`)
```

Comment:
IIUC this fix is only for DataFrame. Series.astype works fine on master?

Reply:
Good point. I tried and `Series.astype` works fine on `master`. Will fix the entry.